irohad/ametsuchi/impl/postgres_options.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ametsuchi/impl/postgres_options.hpp"
+#include <boost/algorithm/string.hpp>
+#include <regex>
+
+namespace iroha {
+  namespace ametsuchi {
+
+    expected::Result<PostgresOptions, std::string> PostgresOptions::create(
+        std::string postgres_options) {
+      // parse options
+      std::regex pattern{R"([^ ]+)"};
```

Comment:
This code doesn't look obvious by itself. Provide comments with some example and parsing on it. And maybe we can use boost::spirit instead of this code?

Reply:
spirit would be less readable for those who are not familiar with it (me for example)

Reply:
It should take less time for people to understand several rules, such as `username=`, `password=`, `dbname=`, accompanied with comments, than the parsing code referenced.

By the way, why not just to use regex to find `dbname=[^ ]+`, and then exclude it from original string?

Reply:
The goal here is to parse all params, not just dbname

Reply:
If it is really critical, I can spend some time to learn spirit and apply it here